src/language-markdown/printer-markdown.js

```diff
@@ -107,7 +107,27 @@ function genericPrint(path, options, print) {
           ? "never"
           : options.proseWrap;
 
-      return printLine(path, node.value, { proseWrap });
+      let isLeadingInsideLinkAlt = false;
+      let isTrailingInsideLinkAlt = false;
+
+      const linkNode = getAncestorNode(path, "link");
+
+      if (linkNode) {
+        isLeadingInsideLinkAlt =
+          linkNode.children.indexOf(parentNode) === 0 && index === 0;
+        isTrailingInsideLinkAlt =
+          linkNode.children.indexOf(parentNode) ===
+            linkNode.children.length - 1 &&
+          index === parentNode.children.length - 1;
```

Comment:
let's give proper names for the equalities in separate variables..

Reply:
It can be difficult to provide meaningful names to these kind of expressions. It's better to leave them as-is.